From 4ac43ddeea128dee2f1ebe2645296731619883cc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthew=20=F0=9F=AA=B7?= <matthew@urgero.net>
Date: Sun, 4 May 2025 01:31:38 -0500
Subject: [PATCH] Update `vFixedSeeds` with new, reliable nodes (#12)

Update `vFixedSeeds` with new nodes that are known and reliable. There
may be additional updates to this list in the future. For now, 4
reliable nodes is sufficient for bootstrapping new nodes.

Also clean out the old, dead DNS seeder nodes and add the new DNS
endpoint for `seed.lotusia.org`. This DNS endpoint is not active due to
issues with the `lotus-seeder` executable and will need to be addressed
in the future.

Rebase provenance: this historical change was adapted onto the pinned
Bitcoin ABC base. Source diff is the recorded Lotus V1 replay commit below.

Historical-source-commit: 74d22f340fee19138a839624d8bd0e96fc71d998
Recorded-replay-commit: b68f3d7ae33efa3d649a19c4229a7c3e77d8e585

diff --git a/contrib/seeds/makeseeds.py b/contrib/seeds/makeseeds.py
index 7cf163f2b9..b0c8109a68 100755
--- a/contrib/seeds/makeseeds.py
+++ b/contrib/seeds/makeseeds.py
@@ -28,7 +28,7 @@ PATTERN_ONION = re.compile(r"^([abcdefghijklmnopqrstuvwxyz234567]{16}\.onion):(\
 
 # Used to only select nodes with a user agent string compatible with the
 # eCash network.
-PATTERN_AGENT = re.compile(r"^(/lotusd:3.(\d+).(\d+)\(.+\)/)")
+PATTERN_AGENT = re.compile(r"^(/lotusd:11.(\d+).(\d+)\(.+\)/)")
 
 
 def parseline(line: str) -> Union[dict, None]:
diff --git a/contrib/seeds/nodes_main.txt b/contrib/seeds/nodes_main.txt
index a50d801b0e..8141244311 100644
--- a/contrib/seeds/nodes_main.txt
+++ b/contrib/seeds/nodes_main.txt
@@ -1,7 +1,2 @@
-35.184.152.63:10605
-45.119.84.253:10605
-65.21.88.223:10605
-116.110.6.191:10605
-123.20.255.166:10605
-152.165.70.246:10605
-173.208.206.181:10605
+147.135.88.232
+147.135.88.233
diff --git a/src/chainparamsseeds.h b/src/chainparamsseeds.h
index d28a4d25a2..6eff9cb512 100644
--- a/src/chainparamsseeds.h
+++ b/src/chainparamsseeds.h
@@ -1,20 +1,15 @@
 #ifndef BITCOIN_CHAINPARAMSSEEDS_H
 #define BITCOIN_CHAINPARAMSSEEDS_H
 /**
- * List of fixed seed nodes for the bitcoin network
+ * List of fixed seed nodes for the Lotus network
  * @generated by contrib/seeds/generate-seeds.py
  *
  * Each line contains a 16-byte IPv6 address and a port.
  * IPv4 as well as onion addresses are wrapped inside an IPv6 address accordingly.
  */
 static SeedSpec6 pnSeed6_main[] = {
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x23,0xb8,0x98,0x3f}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x2d,0x77,0x54,0xfd}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x41,0x15,0x58,0xdf}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x74,0x6e,0x06,0xbf}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x7b,0x14,0xff,0xa6}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x98,0xa5,0x46,0xf6}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0xad,0xd0,0xce,0xb5}, 10605}
+    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x93,0x87,0x58,0xe8}, 10605},
+    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x93,0x87,0x58,0xe9}, 10605}
 };
 
 static SeedSpec6 pnSeed6_test[] = {
diff --git a/src/seeder/options.cpp b/src/seeder/options.cpp
index 8d7ce3d5e3..6d4b174c1a 100644
--- a/src/seeder/options.cpp
+++ b/src/seeder/options.cpp
@@ -106,6 +106,7 @@ int CDnsSeedOpts::ParseCommandLine(int argc, const char **argv) {
         filter_whitelist.insert(NODE_NETWORK | NODE_BLOOM);
         filter_whitelist.insert(NODE_NETWORK_LIMITED);
         filter_whitelist.insert(NODE_NETWORK_LIMITED | NODE_BLOOM);
+        filter_whitelist.insert(NODE_NETWORK | NODE_BLOOM | NODE_NETWORK_LIMITED);
     }
     return CONTINUE_EXECUTION;
 }
